frontend/subscriptions/index.js

```diff
@@ -0,0 +1,46 @@
+import {
+  main$,
+  appDidStart$,
+  navigate$,
+  historyPush,
+  getCurrentRoute,
+  INDEX_PATH,
+} from '@shopgate/engage/core';
+import { logger } from '@shopgate/pwa-core/helpers';
+import { getSelection } from '../selectors';
+import { makeGetSelection } from '../actions';
+
+export default (subscribe) => {
+  const setSwitchSelection$ = main$.filter(({ action }) => action.type === 'SET_SWITCH_SELECTION');
+  const indexDidEnter$ = navigate$.filter(({ action }) => action.params.pathname === INDEX_PATH);
+
+  subscribe(indexDidEnter$, ({ dispatch, getState }) => {
+    const state = getState();
+    const selection = getSelection(state);
+    const route = getCurrentRoute(state);
+
+    if (selection.path !== route.pathname) {
+      dispatch(historyPush({
+        pathname: selection.path,
+      }));
+    }
+  });
+
+  subscribe(setSwitchSelection$, ({ dispatch, getState }) => {
+    const selection = getSelection(getState());
+
+    dispatch(historyPush({
+      pathname: selection.path,
+    }));
+  });
+
+  subscribe(appDidStart$, ({ dispatch, getState }) => {
+    const selection = getSelection(getState());
+
+    try {
+      dispatch(makeGetSelection(selection));
```

Comment:
this logic here causes that users are redirected to the "selection" which was selected when the app was closed the last time.

that's because here a pipeline request is dispatched to get the persisted selection from before. after the response comes in, `setSwitchSelection` is dispatched which causes a redirect to the selection (previous subscription).

so it's not possible to enter the app on any other page than the one from the selection, and i also guess that there will be issues when the app is opened via a push notification that opens e.g. a category or cms page.